### PR TITLE
BKNDLSS-28512 Badge

### PR DESCRIPTION
**Pull request by serhmel**

### CODE REVIEW

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-badge-component/styles/index.less [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+@bl-customComponent-badge-padding: 14px;
+
+@bl-customComponent-badge-badgeBackgroundColor: @bl-customComponent-badge-themeColor;
+@bl-customComponent-badge-badgeTextColor: contrast(@bl-customComponent-badge-badgeBackgroundColor);
+@bl-customComponent-badge-badgeFontSize: 12px;
+@bl-customComponent-badge-badgePadding: 4px;
+
+@bl-customComponent-badge-content-padding: 0;
+@bl-customComponent-badge-content-fontSize: 24px;
+@bl-customComponent-badge-content-color: @bl-customComponent-badge-textColor;
```
[the remark this answers is not in the document]
Comment (serhmel) [reply components/bl-badge-component/styles/index.less:14]: It is used in @bl-customComponent-badge-content-iconColor and @bl-customComponent-badge-content-textColor variables.

File: components/bl-badge-component/src/badge.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    badgeFontSize,
+    badgeForm,
+    badgeAlignment,
+    badgeLabelColor,
+    badgeBackgroundColor,
+    badgeWidth,
+    badgeHeight,
+  } = component;
+  
+  const badgeForms = {
```
Comment (v-excelsior) [inline components/bl-badge-component/src/badge.js:17]: Constant should be outside of component or in external file

Comment (v-excelsior) [reply components/bl-badge-component/src/badge.js:17]: Also types map should be in uppercase.

BadgeForms
BadgeRadiusMap

File: components/bl-badge-component/component.json [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    },
+    {
+      "name": "badgeBackgroundColor",
+      "label": "Badge Background Color",
+      "type": "text"
+    },
+    {
+      "name": "badgeFontSize",
+      "label": "Badge Font Size",
+      "type": "number"
```
Comment (v-excelsior) [inline components/bl-badge-component/component.json:33]: Is it still number? I believe after our disscussion about rem/em/% there should be a `text` type

File: components/bl-badge-component/src/content/image.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+
+export function Image({ imageUrl, imageWidth }) {
+  const styles = useMemo(() => ({
```
Comment (Valodya) [inline components/bl-badge-component/src/content/image.js:4]: as far as I know there is no reason to wrap style with useMemo if they go to a native DOM component

File: components/bl-badge-component/src/content.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { Icon } from './content/icon';
```
Comment (v-excelsior) [inline components/bl-badge-component/src/content.js:1]: Create an index with reexporting all types of content and here will be

import { Icon, Image, Text } from './content';

File: components/bl-badge-component/component.json [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    },
+    {
+      "name": "badgeForm",
+      "label": "Badge Form",
+      "type": "text",
+      "defaultValue": "default"
+    },
+    {
+      "name": "icon",
+      "label": "Icon",
```
Comment (v-excelsior) [inline components/bl-badge-component/component.json:69]: In readme we should describe where we could get  a type of icons.

File: components/bl-badge-component/component.json [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      "type": "text"
+    },
+    {
+      "name": "text",
```
Comment (v-excelsior) [inline components/bl-badge-component/component.json:73]: text -> textContent ?

Too general name for property

**Review by v-excelsior — APPROVED** [review]

File: components/bl-badge-component/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/dist
```
Comment (v-excelsior) [inline components/bl-badge-component/.gitignore:1]: could be removed

File: components/bl-badge-component/component.json [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+    {
+      "name": "badgeFontSize",
+      "label": "Badge Font Size",
+      "type": "text",
+      "showInSettings": true,
+      "hasLogicHandler": false
+    },
+    {
+      "type": "select",
+      "name": "badgeAlignment",
```
Comment (v-excelsior) [inline components/bl-badge-component/component.json:45]: I think as we haven't any other Aligments, we could rename from `badgeAlignment` to `aligment`, at least in code(handler names)

Same for other properties

File: components/bl-badge-component/src/index.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+export default function BadgeComponent({ component, eventHandlers }) {
+  const { display, classList, padding } = component;
+
+  if (!display) {
+    return null;
+  }
+  
+  return (
+    <div className={ 'bl-customComponent-badge ' + classList.join(' ') } style={{ padding }}>
```
Comment (v-excelsior) [inline components/bl-badge-component/src/index.js:12]: use `cn`

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-badge-component/src/badge.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  const styles = {
+    color       : badgeLabelColor,
+    background  : badgeBackgroundColor,
+    borderRadius: BadgeRadiusMap[badgeForm],
+    fontSize    : badgeFontSize,
+    width       : badgeWidth,
+    height      : badgeHeight,
+  };
+
+  if (!badgeVisibility) {
```
Comment (Valodya) [inline components/bl-badge-component/src/badge.js:36]: can we use `display` instead?

**Pull request merged**